src/components/TransactionHistoryRows/TransactionContract.tsx

```diff
@@ -35,18 +39,31 @@ export const TransactionContract = ({
     isReceive && !txDetails.tx.data?.method ? txDetails.tradeTx?.type : txDetails.tx.data?.method
   const isFirstAssetOutgoing = interactsWithWithdrawMethod && isSend
 
+  const isRevoke = useMemo(
+    () =>
+      Boolean(
+        txMetadata?.method === 'approve' && txMetadata.value && bn(txMetadata.value).isZero(),
+      ),
+    [txMetadata],
+  )
   const titlePrefix = translate(
-    txDetails.tx.data?.parser
-      ? `transactionRow.parser.${txDetails.tx.data?.parser}.${i18n}`
-      : 'transactionRow.unknown',
+    (() => {
+      if (txDetails.tx.data?.parser) {
+        return `transactionRow.parser.${txDetails.tx.data?.parser}.${isRevoke ? 'revoke' : i18n}`
+      }
+      return 'transactionRow.unknown'
+    })(),
   )
```

Comment:
A ternary might be nicer:

```ts
const titlePrefix = translate(
  (() => {
    return txDetails.tx.data?.parser
      ? `transactionRow.parser.${txDetails.tx.data?.parser}.${isRevoke ? 'revoke' : i18n}`
      : 'transactionRow.unknown'
  })(),
)
```

Reply:
Nice catch! Added in 723e9216b0097677587e3158fd08716f907db7c6